endorser/main.py

```diff
@@ -1,12 +1,62 @@
-import asyncio
+import os
+from contextlib import asynccontextmanager
 
-from endorser.endorser_processor import listen_endorsement_events
+from dependency_injector.wiring import Provide, inject
+from fastapi import Depends, FastAPI, HTTPException
+
+from endorser.services.dependency_injection.container import Container, get_container
+from endorser.services.endorsement_processor import EndorsementProcessor
 from shared.log_config import get_logger
 
-logger = get_logger("endorser.main")  # override as __name__ gets passed as __main__
+logger = get_logger(__name__)
+
+
+@asynccontextmanager
+async def app_lifespan(_: FastAPI):
+    logger.info("Endorser Service startup")
+
+    # Initialize the container
+    container = get_container()
+    container.wire(modules=[__name__])
+
+    # Start singleton services
+    container.redis_service()
+
+    endorsement_processor = await container.endorsement_processor()
+    endorsement_processor.start()
+
+    yield
+
+    logger.info("Shutdown Endorser services")
+    await endorsement_processor.stop()
```

Comment:
Something prevents endorser service from restarting gracefully (e.g. with watchfile changes, it gets stuck in shutdown)